File: FinalProjSubmission/HS_Dataset/fibAccum.hs
import Text.Printf (printf)

fibAccum :: Int -> Int
fibAccum n = fibHelper n 0 1
  where
    fibHelper 0 a _ = a
    fibHelper n a b = fibHelper (n-1) b (a+b)

main = do
    printf "%d\n" $ fibAccum 10
    printf "%d\n" $ fibAccum 5
    printf "%d\n" $ fibAccum 15

File: FinalProjSubmission/HS_Output/fibAccum.cpp
:
<|im_end|>
<|im_start|>assistant
```cpp
int fibHelper(int n, int a, int b){
  if(n == 0) return a;
  return fibHelper(n-1, b, a+b);
}
int fibAccum(int n){
  return fibHelper(n, 0, 1);
}
int main(){
  std::cout << fibAccum(10) << std::endl;
  std::cout << fibAccum(5) << std::endl;
  std::cout << fibAccum(15) << std::endl;
}
```<|im_end|>